app/lib/big_query/request_event.rb

```diff
@@ -0,0 +1,52 @@
+module BigQuery
+  class RequestEvent
+    def initialize
+      @event_hash = {
+        environment: ENV["RAILS_ENV"],
+        occurred_at: Time.zone.now.iso8601,
+        event_type: "web_request",
+      }
```

Comment:
is it worth `yielding self if block_given?` then you could do:

```
request_event = BigQuery::RequestEvent.new |event|
  event.with_request_details(request)
  event.with_response_details(response)
  event.with_user(current_user)
end
```

Just wondering if it'll help avoid always having to return the instance for each method definition?

Reply:
I've updated with that suggestion @darokel 👍